### ISSUE TITLE
Doesn't correctly clock-stretch before ACK

### ISSUE THREAD

**Issue by nigelorr** [dgatf/I2C-slave-multi-address-RP2040#3]
While emulating a couple of parts (TCA6408, ADS7828), I found the library seemed to be unreliable, sometimes would work perfectly, then fail:

![0x21_data_write_OK_read_fails](https://github.com/dgatf/I2C-slave-multi-address-RP2040/assets/812323/037cc985-ed74-429b-884e-8ca9d80dfffb)

![0x48_3_fails_then_success](https://github.com/dgatf/I2C-slave-multi-address-RP2040/assets/812323/ae320916-2859-476d-9f56-02fcf61a99f5)

**Comment by nigelorr** [comment]
This appears to be because it wasn't correctly clock-stretching the SCL in the part of do_ack_program_instructions that is executed after the byte is received. 

I changed instructions 0-3 to remove the NOP and add an explicit SCL low:
```

static const uint16_t do_ack_program_instructions[] = {
            //     .wrap_target
    0x2021, //  0: wait   0 pin, 1                   
    0xe083, //  1: set    pindirs, 3
    0xef00, //  2: set    pins, 0                   [15]   // Drive SCL low for clock stretching and SDA low to force ACK
    0xc020, //  3: irq    wait 0  

```
I hope that is a useful fix, it has made my testing much more consistent. I presumed from the 'nop' that was there that maybe this had been there previously and removed for some reason, so it might need another solution for wider use.

**Comment by dgatf** [comment]
Fixed with #5
